scripts/tests/test_s3_migrate_to_external_accounts.py

```diff
@@ -0,0 +1,196 @@
+
+    def tearDown(self):
+        self.mock_user_info.stop()
+        super(TestS3Migration, self).tearDown()
+        database['addons3nodesettings'].remove()
+        database['addons3usersettings'].remove()
+        database['s3nodesettings'].remove()
+        database['s3usersettings'].remove()
+        database['externalaccount'].remove()
+
+    def test_migrate_to_external_account(self):
+        assert_equal(ExternalAccount.find().count(), 0)
+        user_settings_document = self.unlinked_user_settings[0]
+        external_account, user, new = migration.migrate_to_external_account(user_settings_document)
+        assert_true(new)
+        assert_equal(ExternalAccount.find().count(), 1)
+        assert_is_not_none(external_account)
+        assert_equal(user_settings_document['owner'], user._id)
+        assert_equal(external_account.provider, 's3')
+        assert_equal(external_account.provider_name, 'Amazon S3')
+        assert_equal(
+            external_account.oauth_key,
+            user_settings_document['access_key']
+        )
+        assert_is_not_none(external_account.display_name)
+
+    def test_make_new_user_settings(self):
+        user_settings_document = self.unlinked_user_settings[0]
+        user = User.load(user_settings_document['owner'])
+        user_settings = migration.make_new_user_settings(user)
+        user.reload()
+        assert(
+            'addons3usersettings' not in user._backrefs['addons']
+        )
+        assert_equal(
+            len(user._backrefs['addons']['s3usersettings']['owner']),
+            1
+        )
+        assert_equal(
+            user._backrefs['addons']['s3usersettings']['owner'][0],
+            user_settings._id
+        )
+        assert_false(hasattr(user_settings, 'access_key'))
+
+    def test_make_new_node_settings(self):
+        node_settings_document = self.node_settings_documents[0]
+        node = Node.load(node_settings_document['owner'])
+        user_settings_document = database['addons3usersettings'].find_one({
+            '_id':  node_settings_document['user_settings']
+        })
+        external_account, user, new = migration.migrate_to_external_account(
+            user_settings_document
+        )
+        user_settings = migration.make_new_user_settings(user)
+        node_settings = migration.make_new_node_settings(
+            node,
+            node_settings_document,
+            external_account,
+            user_settings
+        )
+        assert(
+            'addons3nodesettings' not in node._backrefs['addons']
+        )
+        assert_equal(
+            len(node._backrefs['addons']['s3nodesettings']['owner']),
+            1
+        )
+        assert_equal(
+            node._backrefs['addons']['s3nodesettings']['owner'][0],
+            node_settings._id
+        )
+
+    def test_migrate(self):
+        migration.migrate(dry_run=False)
+        assert_equal(
+            S3UserSettings.find().count(),
+            6
```

Comment:
Instead of hard-coding the number can you use the corresponding variables instead? It corresponds to the length of `self.linked_user_settings` +`self.unlinked_user_settings`, right?